subsys/shell/shell.c

```diff
@@ -1452,6 +1452,16 @@ int shell_start(const struct shell *sh)
 	z_cursor_next_line_move(sh);
 	state_set(sh, SHELL_STATE_ACTIVE);
 
+	/*
+	 * If the shell is stopped with the shell_stop function, its backend remains active
+	 * and continues to buffer incoming data. As a result, when the shell is resumed,
+	 * all buffered data is processed, which may lead to the execution of commands
+	 * received while the shell was stopped.
+	 */
+	if (sh->iface->api->flush) {
+		sh->iface->api->flush(sh->iface);
+	}
```

Comment:
I had also considered implementing it the way you're suggesting, and I’m not insisting on my current approach. The only advantages I see in adding it to the API are that it could handle some backends more quickly, and the flush function might be useful in the future. However, as you mentioned, I don't see any major objections to implementing a unified solution at this point.